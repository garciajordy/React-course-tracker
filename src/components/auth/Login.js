import React, { useState } from 'react'
import axios from 'axios';
import { useHistory } from 'react-router-dom';
import store from "../../store/Store"
import logIn from "../../actions/LoginAction";

function LoginPanel() {
    let history = useHistory()
    const [data, setData] = useState({
        email: '',
        password: '',
        loginErrors: ''
    })
    function handleSubmit(event) {
        axios.post("http://localhost:3001/sessions", {
            user: {
                email: data.email,
                password: data.password
            }
        },
            { withCredentials: true }
        ).then(response => {
            if (response.data.status === "created") {
                store.dispatch(logIn(response.data.user))
                history.push("/dashboard")
                
            } else {
                setData((prev => ({
                    ...prev,
                    loginErrors: "Invalid email or password."
                })))
        }
        }).catch(error => {
            console.log("login error", error)
            setData((prev => ({
                ...prev,
                loginErrors: "Invalid email or password."
            })))
           
        })
        event.preventDefault();
    }

    function handleChange(e) {
        setData((prev => ({...prev, [e.target.name]: e.target.value})))
    }

    return (
        <div>
            <form onSubmit={handleSubmit} >
               <p> {data.loginErrors}</p>
            <input type='email' name='email' placeholder='Email' value={data.email} onChange={handleChange} required />
            <input type='password' name='password' placeholder='Password' value={data.password} onChange={handleChange} required />
                <button type="submit">Login</button>
            </form>
        </div>
    )
}

export default LoginPanel;

