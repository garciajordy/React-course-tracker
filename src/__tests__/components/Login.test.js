import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Login from '../../components/auth/Login';
import store from '../../store/Store';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<Provider store={store}><Login /></Provider>);
  expect(asFragment(<Provider store={store}><Login /></Provider>)).toMatchSnapshot();
});

it('displays heading', () => {
  render(<Provider store={store}><Login /></Provider>);
  expect(screen.getByText('Login')).toBeTruthy();
});
