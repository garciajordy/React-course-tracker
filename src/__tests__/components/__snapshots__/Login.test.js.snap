// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <p>
         
      </p>
      <input
        name="email"
        placeholder="Email"
        required=""
        type="email"
        value=""
      />
      <input
        name="password"
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</DocumentFragment>
`;
